src/components/Card/Card.js

```diff
@@ -3,18 +3,27 @@ import { Link } from 'react-router-dom';
 import styles from './card.module.css';
 
 const Card = ({ product }) => {
-  const { href, title, image, alt, price } = product;
+  const { href, title, image, price } = product;
+
+  let maxTitleLength = 15;
+  if (window.innerWidth >= 630 && window.innerWidth <= 1023) {
+    maxTitleLength = 12;
+  } else if (window.innerWidth >= 1023) {
+    maxTitleLength = 19;
+  }
+
+  const trimmedTitle = title.length > maxTitleLength ? `${title.slice(0, maxTitleLength)}...` : title;
```

Comment:
you can do it much simpler and clearly by using `text-overflow: ellipsis;` style. https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow

Reply:
Done